### ISSUE TITLE
setDimensions bug? ul.slider-list too tall after componentDidMount

### ISSUE THREAD

**Issue by omasback** [FormidableLabs/nuka-carousel#202]
My horizontal slider with all default props is too tall on first mount, and I think this is why:

* `componentWillMount -> setInitialDimensions` sets `ul.slider-list` and each `li.slider-slide` to zero width

* Then we come to `componentDidMount -> setDimensions`, where

* `firstSlide.style.height = 'auto';` allows the `li.slider-slide` to grow vertically 

* which is then measured to calculate the height of the slider 
`slideHeight = firstSlide.offsetHeight;`

**BUT**

* `li.firstSlide` is still zero width! So all of its content gets squeezed to minimal width, forcing it down and making it way too tall!

* A few lines later, `slideWidth` is properly calculated. 
`slideWidth = frame.offsetWidth / props.slidesToShow * props.slideWidth;`
But it's not applied to the DOM yet, so we still wouldnt be able to properly measure the `firstSlide.offsetHeight`

* Then `setState` triggers a render which applies the correct width to `ul.slider-list` and `li.slider-slide`, but the **wrong height** to `ul.slider-list`. 

* Then, if we're lucky enough to have mounted this component on first page load, we may have a `readystatechange` event trigger another `setDimensions` which will calculate the correct `firstSlide.offsetHeight`, this time because the correct `slideWidth` has already been applied to the DOM.

So basically `setDimensions` needs to run twice *after* mounting in order to get it right. And if I'm mounting this component in a SPA long after readystatechange has fired, it only runs once. Am I doing something wrong? Or is this a bug. I'm not sure what the fix should be because I'm afraid that fixing for my own case would break other configurations. Whatever the case, **the first slide needs to be given its final width in the DOM before its offsetHeight is measured.**

**Comment by raduflp** [comment]
I'm having the same issue. @omasback did you find a fix/workaround for that?

**Comment by sofiekedvik** [comment]
I'm also having the same issue :( anyone has a fix for this yet?

**Comment by antoniocosentino** [comment]
same issue here, I still haven't found a "clean" solution

**Comment by omasback** [comment]
Sorry all, the fix I settled on was to use a different carousel.

**Comment by EvgeniyRRU** [comment]
@omasback, can you advise, which one?

**Comment by Morhaus** [comment]
Here's how I fixed this issue for my particular case. Use this component as you would the `nuka-carousel` `Carousel` component.

`Carousel.jsx`
```
import { Component } from "react";
import NCarousel from "nuka-carousel";

export default class Carousel extends Component {
  state = {
    width: 0
  };

  componentDidMount() {
    this.setState(s => ({ width: this.ref.offsetWidth }));
  }

  render() {
    return (
      <>
        <div
          ref={ref => {
            this.ref = ref;
          }}
        />
        {this.state.width != 0 && (
          <NCarousel
            {...this.props}
            initialSlideWidth={this.state.width}
          </NCarousel>
        )}
      </>
    );
  }
}
```

**Comment by ashika01** [comment]
@omasback 
It's easier to show us than tell us what's going wrong with your code. Because of this, we ask that you do one of the following to help us reproduce the bug:
- Create a public minimal repository that we can `git clone`, with install + error reproduction steps in the README.
- Reproduce your issue with a sample sandbox and paste the link here.

**Comment by omasback** [comment]
Might want to reopen this one, since that fix was apparently reverted.

**Comment by willmoss1000** [comment]
You can get around this by manually setting the height of your slides divs to a small value i.e. `style={{height: 100}}`

**Comment by haugan** [comment]
> 
> 
> You can get around this by manually setting the height of your slides divs to a small value i.e. `style={{height: 100}}`

Not working for me.

**Comment by haugan** [comment]
> 
> 
> You can get around this by manually setting the height of your slides divs to a small value i.e. `style={{height: 100}}`

Not working for me.

**Comment by vytenis-nfq** [comment]
Try triggering the resize event on document

**Comment by sarmeyer** [comment]
fixed with #469
